src/wymeditor/editor/base.js

```diff
@@ -33,29 +34,17 @@ WYMeditor.editor.prototype._init = function () {
         sContainer,
         oContainer;
 
-    if (WYMeditor.isInternetExplorerPre11()) {
-        WymClass = new WYMeditor.WymClassTridentPre7(wym);
-    } else if (WYMeditor.isInternetExplorer11OrNewer()) {
-        WymClass = new WYMeditor.WymClassTrident7(wym);
-    } else if (jQuery.browser.mozilla) {
-        WymClass = new WYMeditor.WymClassGecko(wym);
-    } else if (jQuery.browser.safari || jQuery.browser.webkit ||
-               jQuery.browser.chrome) {
-        if (jQuery.browser.version === '537.36') {
-            // This seems to indicate Blink. See:
-            // https://stackoverflow.com/questions/20655470
-            //WymClass = new WYMeditor.WymClassBlink(wym);
-            // For now we use the WebKit editor class in Blink.
-            WymClass = new WYMeditor.WymClassWebKit(wym);
-        } else {
-            WymClass = new WYMeditor.WymClassWebKit(wym);
-        }
-    }
+    // Get the constructor for the browser-specific instance
+    WymClass = WYMeditor._getWymClass();
 
-    if (WymClass === false) {
+    if (!WymClass) {
+        // We don't support this browser. Don't initialize.
```

Comment:
It feels like that if we can't reliably print to the console via `console.warn` to give a developer some feedback, we should throw an exception. This silent failure would be really frustrating behavior for someone to debug.